### ISSUE TITLE
Angular's Event type in conflict with dart:html and lib.d.ts Event types

### ISSUE THREAD

**Issue by yjbanov** [angular/angular#4322]
We should rename it. There's too much code that uses both DOM APIs and Angular 2. It would be unreasonable to have users rename/hide or otherwise manually resolve these two types.

NOTE: also have to rename the `events` property in `Directive`.

**Comment by mhevery** [comment]
Do you have a suggestion for a good name?

**Comment by yjbanov** [comment]
Dunno. How about this?

```dart
@Emits('rainbow')
var rainbows = new EventEmitter();
```

**Comment by mhevery** [comment]
@vsavkin @pkozlowski-opensource @IgorMinar @vicb can you comment?

**Comment by pkozlowski-opensource** [comment]
I like the current `@Event` but I understand the issue on the Dart side. `@Emits` sounds like a good name but it is not consistent with other names (`@Property`, `@HostBinding`) which are nouns.

Having said this I don't have great name candidates. The only thing that comes to my mind is `@CustomEvent` or `@DirectiveEvent` if we want to stick with nouns. `@Emitter` sounds too confusing...

**Comment by mhevery** [comment]
@pkozlowski-opensource I think `@Property` is declaring what it is. So noun here is fine.  Where as event is an action so `Emits` sounds nice.

**Comment by pkozlowski-opensource** [comment]
I can definitively live with `Emits`

**Comment by vicb** [comment]
Emits sounds good

**Comment by PascalPrecht** [comment]
Based on @mhevery 's arguments, I'd also vote for `@Emits`. It's very declarative, which is nice.

**Comment by IgorMinar** [comment]
Emits is a verb, all the other decorators are nouns. We should be
consistent about this.

**Comment by robwormald** [comment]
what about `@Events(['click', 'drag'])` - avoids the conflict, keeps the noun, prevents boilerplate when there are lots of events?

Note that `EventEmitter` will likely be renamed to ____? and will have an `emit()` method on it - see https://github.com/angular/angular/issues/4287

**Comment by mhevery** [comment]
@IgorMinar Why do you think verb/noun distinction is so important, given that you are declaring an action? What you are declaring is a verb and you will use a verb.

@naomiblack @bradlygreen do you want to comment?

**Comment by yjbanov** [comment]
We should postpone this till after #3605 lands. Marking as blocked.

**Comment by mhevery** [comment]
It already landed: https://github.com/angular/angular/commit/adbfd29fd761135d51985564edcb4db7f8b6a26a
